client/http/url.c

```diff
@@ -0,0 +1,284 @@
+#include "url.h"
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+static char *str_hosttype[] = { "host ipv4", "host ipv6", "host domain", NULL };
+
+//static char *strndup(const char *str, int n)
+//{
+//	char *dst;
+//	if (!str) return NULL;
+//	if (n < 0) n = strlen(str);
+//	if (n == 0) return NULL;
+//	if ((dst = (char *)malloc(n + 1)) == NULL)
+//	{
+//		return NULL;
+//	}
+//
+//	memcpy(dst, str, n);
+//	dst[n] = 0;
+//	return dst;
+//}
+
+static int host_is_ipv4(char *str)
+{
+	if (!str) return 0;
+	while (*str)
+	{
+		if ((*str >= '0' && *str <= '9') || *str == '.')
+			str++;
+		else
+			return 0;
+	}
+	return 1;
+}
+
+void parse_query(url_field_t *url, char *query);
+void parse_query(url_field_t *url, char *query)
```

Comment:
You declare function and define it immediately. Why do you do it?

Reply:
If this function is used only in that file you can use static modifier.

Reply:
Move the declare above rather than use static since static modifier has different use case.